### DISCUSSION TITLE
PromptConfiguration - RGB in default or not?

### DISCUSSION THREAD

**Discussion by kindermannhubert** [waf/PrettyPrompt#59] [General]
Should we use RGB AnsiColors for default values in ```PromptConfiguration``` or not?

If not then we could add optional ```bool allowRgbDefaults``` in its ctor. Then there would be two default value sets - one nonRGB (ugly) and one nicer. But then there is question if ```allowRgbDefaults``` should be optional and if so what default value it should have :)

**Comment by waf** [comment]
I'd prefer using a standard terminal AnsiColor for defaults, rather than RGB defaults. I think it's surprising for users if we choose an arbitrary RGB value that happens to look good on our system -- better to use the ANSI color slots as they were intended to be used.

I just looked at a few other projects to get another opinion:

- terminal-kit (popular project from JS) defaults to ANSI colors (e.g. in [this example](https://github.com/cronvel/terminal-kit/blob/master/doc/high-level.md#progressbar-options-)).
- Spectre.Console (probably the most popular console output library for .NET?) uses ANSI colors. [e.g. in their example code](https://github.com/spectreconsole/spectre.console/blob/main/examples/Console/Borders/Program.cs).

As long as the user can easily provide RGB colors in the configuration object, I think it's fine to default to ANSI colors.
